app/bundles/LeadBundle/Tests/EventListener/ReportSubscriberTest.php

```diff
@@ -1034,4 +1043,104 @@ public function testOnReportDisplay(string $event): void
         $this->reportSubscriber->onReportBuilder($this->reportBuilderEventMock);
         $this->reportSubscriber->onReportDisplay($this->reportDataEventMock);
     }
+
+    public function testOnReportColumnCollectForCompany(): void
+    {
+        $companyFields  = [
+            'comp.id'   => [
+                'alias' => 'comp_id',
+                'label' => 'mautic.lead.report.company.company_id',
+                'type'  => 'int',
+                'link'  => 'mautic_company_action',
+            ],
+            'companies_lead.is_primary' => [
+                'label' => 'mautic.lead.report.company.is_primary',
+                'type'  => 'bool',
+            ],
+            'companies_lead.date_added' => [
+                'label' => 'mautic.lead.report.company.date_added',
+                'type'  => 'datetime',
+            ],
+        ];
+
+        $fields         = [
+            'comp.id'   => [
+                'alias' => 'comp_id',
+                'label' => 'mautic.lead.report.company.company_id',
+                'type'  => 'int',
+                'link'  => 'mautic_company_action',
+            ],
+        ];
+
+        $columnCollectEventMock = $this->createMock(ColumnCollectEvent::class);
+
+        $columnCollectEventMock->expects($this->once())
+            ->method('getObject')
+            ->willReturn('company');
+
+        $this->companyReportDataMock->expects($this->once())
+            ->method('getCompanyData')
+            ->willReturn($companyFields);
+
+        $columnCollectEventMock->expects($this->once())
+            ->method('addColumns')
+            ->with($fields);
```

Comment:
Please do not mock DTO objects like events
```suggestion
        $columnCollectEvent = new ColumnCollectEvent('company');

        $this->companyReportDataMock->expects($this->once())
            ->method('getCompanyData')
            ->willReturn($companyFields);
```

Later on you can call to assert that the columns are set correctly.
```php
Assert::assertSame($fields, $columnCollectEvent->getColumns());
```
This is happening in more places in this PR. The reason is that there is no need to mock objects that have no dependencies on other objects. Just use those objects directly. And you'll test the real behavior instead of mocked one. And it will increase your test coverage.